### PR TITLE
Implement snapshot purge

### PR DESCRIPTION
**Pull request by shuo-wu**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#7199

#### What this PR does / why we need it:

Need to rely on purge APIs to clean up system snapshots created by rebuilding automatically

### CODE REVIEW

File: pkg/spdk/replica.go [hunk 1]
```diff
@@ -1071,6 +1071,51 @@ func (r *Replica) SnapshotRevert(spdkClient *spdkclient.Client, snapshotName str
...
+
+	if r.ChainLength < 2 {
+		return fmt.Errorf("invalid chain length %d for replica snapshot purge", r.ChainLength)
+	}
+
+	// delete all non-user-created snapshots
+	for snapshotLvolName, snapSvcLvol := range r.SnapshotLvolMap {
+		if snapSvcLvol.UserCreated {
+			continue
+		}
```
Comment (derekbit) [inline pkg/spdk/replica.go:1102]: We have a [bug](https://github.com/longhorn/longhorn/issues/9054#issuecomment-2245049025) in v1.6.2.
The `userCreated` of snapshots for v2 volumes created in v1.6.2 are set to `false`.
The purge will lead to these snapshots being deleted.
We can add this in the `important note`? Is it OK? cc @shuo-wu @DamiaSan @innobead

Comment (shuo-wu) [reply pkg/spdk/replica.go:1102]: YES. I am fine with it.

File: pkg/types/types.go [hunk 2]
```diff
@@ -71,4 +71,7 @@ const (
 	ProgressStateComplete   = "complete"
 	ProgressStateInProgress = "in_progress"
 	ProgressStateStarting   = "starting"
+
+	// SPDKShallowCopyStateNew is the state returned from spdk_tgt. There is no underscore in the string.
+	SPDKShallowCopyStateInProgress = "in progress"
```
Comment (derekbit) [inline pkg/types/types.go:76]: It is annoying..

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
@shuo-wu It is a DRAFT. Will you convert it to READY?

**Review by derekbit — DISMISSED** [review]

**Comment by derekbit** [comment]
@shuo-wu will update the unit test later.

**Review by derekbit — APPROVED** [review]

**Pull request merged**